vyper/signatures/interface.py

```diff
@@ -151,10 +151,7 @@ def extract_interface_str(code, contract_name, interface_codes=None):
     for idx, event in enumerate(events):
         if idx == 0:
             out += "# Events\n\n"
-        out += "{event_name}: event({{{args}}})\n".format(
-            event_name=event.name,
-            args=", ".join([arg.name + ": " + str(arg.typ) for arg in event.args])
-        )
+        out += f"{event.name}: event({{{', '.join([arg.name + ': ' + str(arg.typ) for arg in event.args])}}})\n"  # noqa: E501
```

Comment:
```suggestion
        event_args_str = ', '.join([arg.name + ': ' + str(arg.typ) for arg in event.args])
        out += f"{event.name}: event({event_args_str})\n"
```

Reply:
Did the original need the other two sets of curly braces for something?

Reply:
Yeah, I'm not sure why that was there, but would be good to follow up on it

Reply:
I ran tests on this change, and they crashed on test_basic_interface - including readout:
```
=================================== FAILURES ===================================
_________________________ test_basic_extract_interface _________________________
[gw1] linux -- Python 3.7.3 /usr/bin/python3

    def test_basic_extract_interface():
        code = """
    # Events
    
    Transfer: event({_from: address, _to: address, _value: uint256})
    
    # Functions
    
    @constant
    @public
    def allowance(_owner: address, _spender: address) -> (uint256, uint256):
        return 1, 2
        """
    
        out = compile_code(code, ['interface'])
        out = out['interface']
        code_pass = '\n'.join(code.split('\n')[:-2] + ['    pass'])  # replace with a pass statement.
    
>       assert code_pass.strip() == out.strip()
E       AssertionError: assert '# Events\n\n...6):\n    pass' == '# Events\n\n...6):\n    pass'
E           # Events
E           
E         - Transfer: event({_from: address, _to: address, _value: uint256})
E         ?                 -                                             -
E         + Transfer: event(_from: address, _to: address, _value: uint256)
E           
E           # Functions...
E         
E         ...Full output truncated (6 lines hidden), use '-vv' to show

code       = '\n# Events\n\nTransfer: event({_from: address, _to: address, _value: uint256})\n\n# Functions\n\n@constant\n@public\ndef allowance(_owner: address, _spender: address) -> (uint256, uint256):\n    return 1, 2\n    '
code_pass  = '\n# Events\n\nTransfer: event({_from: address, _to: address, _value: uint256})\n\n# Functions\n\n@constant\n@public\ndef allowance(_owner: address, _spender: address) -> (uint256, uint256):\n    pass'
out        = '# Events\n\nTransfer: event(_from: address, _to: address, _value: uint256)\n\n# Functions\n\n@constant\n@public\ndef allowance(_owner: address, _spender: address) -> (uint256, uint256):\n    pass\n\n'

tests/parser/functions/test_interfaces.py:38: AssertionError

```

Reply:
It needs all three sets of curly braces to pass, though I don't know why. I'll update the suggestion:
```suggestion
event_args_str = ', '.join([arg.name + ': ' + str(arg.typ) for arg in event.args])
out += f"{event.name}: event({{{event_args_str}}})\n"
```

Reply:
Ah, well one set is needed because you specify events as `event_name: event({arg1: typ1, ...})`, the other is probably there to escape it? 2 braces might work, maybe.

Reply:
I tried 2, and it failed.

Reply:
Super weird... 3 it is!